src/commands.js

```diff
@@ -0,0 +1,157 @@
+
+const chalk = require('chalk')
+
+const repoVersion = require('./repo/version')
+const migrator = require('./index')
+const templates = require('./migration-templates')
+const migrations = require('../migrations')
+
+function asyncClosure (fnc) {
+  return function asyncWrapper ({ resolve, ...options }) {
+    resolve(fnc(options))
+  }
+}
+
+function reportingClosure (action) {
+  return (migration, currentlyMigrated, totalToMigrate) =>
+    process.stdout.write(`${chalk.green(`[${currentlyMigrated}/${totalToMigrate}]`)} Successfully ${action} ${chalk.bold(migration.version)}: ${migration.description}\n`)
+}
+
+async function migrate ({ repoPath, migrations, to, dry, revertOk }) {
+  repoPath = repoPath || process.env.IPFS_PATH || path.join(os.homedir(), '.jsipfs')
+
+  // Import migrations if set
+  if (migrations) {
+    migrations = require(migrations)
+  }
+
+  const version = await repoVersion.getVersion(repoPath)
+
+  let action
+  if (dry) {
+    action = 'loaded migration'
+  } else {
+    if (!to || (version <= to)) {
+      action = 'migrated to version'
+    } else {
+      action = 'reverted version'
+    }
+  }
+
+  const options = {
+    migrations: migrations,
+    toVersion: to,
+    ignoreLock: false,
+    onProgress: reportingClosure(action),
+    isDryRun: dry
+  }
+
+  if (!to || (version <= to)) {
+    await migrator.migrate(repoPath, options)
+  } else if (revertOk) {
+    await migrator.revert(repoPath, to, options)
+  } else {
+    throw new Error('The migration would require reversion of the repo, but you have not allowed it as \'--revert-ok\' is not present.')
+  }
+}
+
+async function status ({ repoPath, migrations }) {
+  repoPath = repoPath || process.env.IPFS_PATH || path.join(os.homedir(), '.jsipfs')
+
+  // Import migrations if set
+  if (migrations) {
+    migrations = require(migrations)
+  }
+
+  const version = await repoVersion.getVersion(repoPath)
+  const lastMigrationVersion = migrator.getLatestMigrationVersion(migrations)
+  const statusString =
+    version < lastMigrationVersion ? chalk.yellow('There are migrations to be applied!') : chalk.green('Nothing to migrate!')
+
+  return `${statusString}\nCurrent repo version: ${version}\nLast migration's version: ${lastMigrationVersion}`
+}
+
+async function getAuthor () {
+  try {
+    const name = (await exec('git config --get user.name')).stdout
+    const email = (await exec('git config --get user.email')).stdout
+    return `${name.replace('\n', '')} <${email.replace('\n', '')}>`
+  } catch (e) {
+    return ''
+  }
+}
+
+async function add ({ empty }) {
+  const newMigrationVersion = migrator.getLatestMigrationVersion() + 1
+  const newMigrationFolder = path.join(__dirname, '..', 'migrations', 'migration-' + newMigrationVersion)
+
+  const migrationsImport = migrations.map((migration) => migration.empty ? `  Object.assign({version: ${migration.version}}, emptyMigration),` : `  require('./migration-${migration.version}'),`)
```

Comment:
Let's make sure migration.version is a number here

Reply:
I don't think that is needed. This generates string so even if `migration.version` would be a string, it would not make any difference.

Reply:
I'm thinking more along the lines of dependency injection issues